geopy/distance.py

```diff
@@ -526,8 +561,118 @@ def destination(self, point, bearing, distance=None): # pylint: disable=W0621
 
         return Point(units.degrees(radians=lat2), units.degrees(radians=lng2))
 
+class geodesic(Distance):
+    """
+    Calculate the geodesic distance between two points
 
-# Set the default distance formula to the most generally accurate.
+    Set which ellipsoidal model of the earth to use by specifying an
+    ``ellipsoid`` keyword argument. The default is 'WGS-84', which is the
+    most globally accurate model.  If ``ellipsoid`` is a string, it is
+    looked up in the `ELLIPSOIDS` dictionary to obtain the major and minor
+    semiaxes and the flattening. Otherwise, it should be a tuple with those
+    values.  See the comments above the `ELLIPSOIDS` dictionary for
+    more information.
+
+    Example::
+
+        >>> from geopy.distance import geodesic
+        >>> newport_ri = (41.49008, -71.312796)
+        >>> cleveland_oh = (41.499498, -81.695391)
+        >>> print(geodesic(newport_ri, cleveland_oh).miles)
+        538.390445368
+    """
 
-distance = VincentyDistance = vincenty
+    ellipsoid_key = None
+    ELLIPSOID = None
+    geod = None
+
+    def __init__(self, *args, **kwargs):
+        if not HAVE_GEODESIC:
+            raise NotImplementedError(
+                "neither pyproj nor geographiclib is available")
+        self.set_ellipsoid(kwargs.pop('ellipsoid', 'WGS-84'))
+        self.method = "pyproj" if HAVE_PROJ else "geographiclib"
```

Comment:
I would suggest to make 2 different implementations of `geodesic` class instead (probably having a common base class): one for `pyproj` and another for `geographiclib`. This way we will get rid of these ugly `if`s below and we will be able to run tests against both implementations (especially `geographiclib` when having `pyproj` installed).